### PR TITLE
fix /tasks endpoint response

### LINKED ISSUE #1391: GET /api/v2/tasks/TASKID should return task flat, with embedded links.
currently the GET `/api/v2/tasks/TASKID` endpoint returns:
```{ "links":{..."}, "task":{ "id":"...", "organizationId":"...", "name":"..." ...}```

should return:
```{"id":"...", "organizationId":"...", "name":"..." ..., "links":{...}}```

### PR DESCRIPTION
**Pull request by zhulongcheng**
Updates #1391 #1392

  - [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergeable
  - [x] Tests pass
  - [x] swagger.json updated (if modified Go structs or API)
  - [ ] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: http/task_service.go [hunk 1]
```diff
@@ -99,21 +99,28 @@ func newTaskResponse(t platform.Task) taskResponse {
...
+		rs.Tasks = append(rs.Tasks, newTaskResponse(*t))
+	}
+	return rs
 }
 
 type runResponse struct {
 	Links map[string]string `json:"links"`
-	Run   platform.Run      `json:"run"`
+
+	// TODO @zhulongcheng
```
Comment (mark-rushakoff) [inline http/task_service.go]: Can you either elaborate the TODO - what is there to do here? - or remove the comment?

File: http/task_service.go [hunk 2]
```diff
@@ -99,21 +99,28 @@ func newTaskResponse(t platform.Task) taskResponse {
...
 }
 
 func newTasksResponse(ts []*platform.Task) tasksResponse {
-	return tasksResponse{
+	rs := tasksResponse{
 		Links: map[string]string{
 			"self": tasksPath,
 		},
-		Tasks: ts,
+		Tasks: make([]taskResponse, 0, len(ts)),
```
Comment (mark-rushakoff) [inline http/task_service.go]:
```suggestion
		Tasks: make([]taskResponse, len(ts)),
```

File: http/task_service.go [hunk 3]
```diff
@@ -99,21 +99,28 @@ func newTaskResponse(t platform.Task) taskResponse {
...
 	}
+
+	for _, t := range ts {
```
Comment (mark-rushakoff) [inline http/task_service.go]:
```suggestion
	for i, t := range ts {
```

File: http/task_service.go [hunk 4]
```diff
@@ -99,21 +99,28 @@ func newTaskResponse(t platform.Task) taskResponse {
...
+		rs.Tasks = append(rs.Tasks, newTaskResponse(*t))
```
Comment (mark-rushakoff) [inline http/task_service.go]:
```suggestion
		rs.Tasks[i] = newTaskResponse(*t)
```

**Comment by zhulongcheng** [comment]
Hi @mark-rushakoff , this is a  Work In Progress PR. 
Sorry, I didn't make the status of this PR clear.
Thank you for your suggestions ❤️ ❤️ ,  I will complete this PR as soon as possible.

**Comment by mark-rushakoff** [comment]
Thanks @zhulongcheng !

**Pull request merged**